### PR TITLE
react native 6 months in

### PR DESCRIPTION
**Pull request by chase-seibert**

### CODE REVIEW

File: _posts/2017-05-26-react-native-six-months-in.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+# Limitations
+
+We have run into a number of areas where we are making compromises on the user
+experience. It's not totally clear yet which of these are legitimate things that
+React Native is not good at, and which are due to us as a team not having
+focused on them.
+
+## Performance
+
+We have seen persistent issues with app performance, particularly around the
```
Comment (sjyoung12) [inline _posts/2017-05-26-react-native-six-months-in.md]: I think the tab-switching latency is actually a direct artifact of `react-native-router-flux`.  The latency is completely gone (on iOS, at least) when using `react-native-navigation`.  Check out the build from [this PR](https://github.com/NerdWallet/mobile-creditcards/pull/918).

To be fair, the latency _might_ still be indicative of some fundamental flaw in how RN renders screens, but I'm not convinced that's the case.

File: _posts/2017-05-26-react-native-six-months-in.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+intentional. It's very tempting once you have a cross platform app to make UX
+decisions to optimize for code re-use, not the best use experience. Even more
+troubling is the times when the engineers themselves don't realize that a
+particular design is not following best practice UX; typically due to many of
+them not having mobile experience.
+
+# Question marks
+
+In my mind, the biggest open question is around how these mobile native
+platforms continue to evolve. Both Android and iOS have there developer
```
Comment (sjyoung12) [inline _posts/2017-05-26-react-native-six-months-in.md]: 'there' => 'their' 
;)

File: _posts/2017-05-26-react-native-six-months-in.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+---
+
+NerdWallet released our first cross platform mobile app at the start of 2017,
+about six months ago. We choose to use React Native, and we are still using
+99% of the same code base for the iOS and Android apps. What have we learned
+so far?
+
+# Initial reasoning
+
+We initially choose React Native because it was the only way to hit our goals. We
```
Comment (S-Bala) [inline _posts/2017-05-26-react-native-six-months-in.md]: choose => chose?

File: _posts/2017-05-26-react-native-six-months-in.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+didn't have any mobile native engineers, and we had been asked to deliver an
```
Comment (S-Bala) [inline _posts/2017-05-26-react-native-six-months-in.md]: 'mobile native' => 'native mobile'

File: _posts/2017-05-26-react-native-six-months-in.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+scrolls down on a list view, you need to write the animation yourself. We
+initially tried the naive solution where we control style properties from
+JavaScript. That did not perform well. But it was relatively easy to use the
+actual bridged animation primitives, at least for simple animations. We still
+don't have a good sense of how difficult complex custom animations will be.
+
+## Fonts
+
+Initially we did not use the correct default fonts for each platform. It turned
+out to be relatively simple to use either San Fransisco or Roboto based on the
```
Comment (S-Bala) [inline _posts/2017-05-26-react-native-six-months-in.md]: 'San Fransisco' => 'San Francisco'

File: _posts/2017-05-26-react-native-six-months-in.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+is a chance that React Native fades away in the coming years as the framework
+ecosystem [continues to evolve](https://trends.google.com/trends/explore?q=react%20native,phonegap,cordova,Xamarin).
+
+![trends](https://dl.dropboxusercontent.com/spa/sffu0th1cc1sg9q/ellfigls.png)
+
+The second largest risk I see is a little more insidious. As we have ramped up
+the team with engineers who have prior mobile experience, they have begun to
+point out areas where we are not following best UX practices. Some of these are
+intentional. It's very tempting once you have a cross platform app to make UX
+decisions to optimize for code re-use, not the best use experience. Even more
```
Comment (S-Bala) [inline _posts/2017-05-26-react-native-six-months-in.md]: 'use experience' => 'user experience'?

File: _posts/2017-05-26-react-native-six-months-in.md (hunk 1 above)
Comment (yloeza) [inline _posts/2017-05-26-react-native-six-months-in.md]: I had the same suspicious that the issue was in the `react-native-router-flux` navigation lib, at the end the logic for rendering each tab content it's on their side, I'm now sure why they chose to render not only the visible tab but all of them, that results in a bunch on `selectors` from the connected containers to been executed when there was no reason. And the cost of a lot of JS execution time wasted.

**Pull request merged**